watertap/property_models/multicomp_aq_sol_prop_pack.py

```diff
@@ -272,41 +272,49 @@ def build(self):
         self.Liq = AqueousPhase()
 
         # list to hold all species (including water)
-        self.component_list = Set()
+        self.component_list = Set(dimen=1)
 
         # components
         self.H2O = Solvent()
 
-        # blank sets
-        self.cation_set = Set()
-        self.anion_set = Set()
-        self.solute_set = Set()
-        self.ion_set = Set()
+        # Other component sets
+        self.solute_set = Set(dimen=1)  # All components except Solvent() ("H2O")
+        self.cation_set = Set(dimen=1)  # Components with charge >0
+        self.anion_set = Set(dimen=1)  # Components with charge <0
+        self.neutral_set = Set(dimen=1)  # Components with charge =0
+        self.ion_set = Set(dimen=1)  # All Ion Components (cations + anions)
 
+        # Group components into different sets
         for j in self.config.solute_list:
+            if j == "H2O":
+                raise ConfigurationError(
+                    "'H2O'is reserved as the default solvent and cannot be a solute."
+                )
+            self.add_component(j, Solute())
             if j in self.config.charge:
                 if self.config.charge[j] == 0:
                     raise ConfigurationError(
                         "The charge property should not be assigned to the neutral component: {}".format(
                             j
                         )
                     )
-                elif self.config.charge[j] > 0:
+                self.del_component(j)
+                if self.config.charge[j] > 0:
+                    self.del_component(j)
                     self.add_component(
-                        str(j),
+                        j,
                         Cation(charge=self.config.charge[j], _electrolyte=True),
                     )
-                    self.component_list.add(str(j))
-                    self.ion_set.add(str(j))
+                    self.ion_set.add(j)
                 else:
+                    self.del_component(j)
                     self.add_component(
-                        str(j),
+                        j,
                         Anion(charge=self.config.charge[j], _electrolyte=True),
                     )
-                    self.component_list.add(str(j))
-                    self.ion_set.add(str(j))
+                    self.ion_set.add(j)
             else:
-                self.add_component(str(j), Solute())
+                self.neutral_set.add(j)
```

Comment:
This logic can be expressed without calling `del_component`.

Reply:
@bknueven , logically I thought the same as you did, but it turned out I had to use the `del_component` to first take the `j` off the Solute() class or the `solute_set` and then add it into the `Cation()` or `Anion()` class.  I think this is a control by Pyomo block and here all sets are children of the same block.  I did deleted a useless `del_component` line and added a test for a error message in the recent push.